### PR TITLE
feat: move notifications origin resolving to backend

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

Moves notifications origin resolving from frontend to backend using the httpAuth service.
Also removed support for old backend TokenManager and IdentityApi as we do not support it for notifications.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/notifications-backend/src/service/router.ts [hunk 1]
```diff
@@ -82,14 +78,10 @@ export async function createRouter(
     catalog ?? new CatalogClient({ discoveryApi: discovery });
   const store = await DatabaseNotificationsStore.create({ database });
 
-  const { auth, httpAuth } = createLegacyAuthAdapters(options);
```
Comment (Rugvip) [inline plugins/notifications-backend/src/service/router.ts:85]: Yep this makes perfect sense, wasn't any point adding it 😅

**Review by Rugvip — APPROVED** [review]
Nice! 🎉

**Comment by drodil** [comment]
Will solve the conflicts tomorrow 👌

**Comment by Rugvip** [comment]
Conflicts looks solved, I :shipit:

**Pull request merged**